### PR TITLE
Idle support

### PR DESCRIPTION
**Pull request by chaoky**
https://github.com/Mstrodl/elcord/issues/33

### CODE REVIEW

**Comment by LennyPhoenix** [comment]
This should probably be merged in

**Comment by Mstrodl** [comment]
@LennyPhoenix I can't merge until the author marks it as not a draft

**Comment by chaoky** [comment]
It's not complete, and I didn't had the time to look into it again. 
(doc strings, variables names, etc)

**Comment by Zulu-Inuoe** [comment]
@chaoky Would you mind if I took a stab at updating this PR a bit? It mostly looks good to me, but I just want to update minor formatting & actually go ahead and turn it on by default with a reasonable time (5 mins?) since I'm willing to bet most people would rather it just work this way ootb (but I'm open to suggestions!)

**Comment by chaoky** [comment]
@Zulu-Inuoe I'm so sorry, your message probably got lost in my notifications a few months ago.
I'll put the default on 5min, any suggestion for the default idle text and icon?

**Comment by chaoky** [comment]
while refactoring the code, I realized that I was relying on a bug to stop the sentinel updates, I made it more explicit, but any suggestion is welcomed

File: elcord.el [hunk 1]
```diff
@@ -589,5 +604,37 @@ If there is no 'previous' buffer attempt to find a non-boring buffer to initiali
...
+(defun elcord--cancel-idle ()
+  "Resume presence update and timer."
+  (unless elcord-quiet
+      (message "elcord: welcome back"))
+  (remove-hook 'pre-command-hook 'elcord--cancel-idle)
+  ;;hacky way to resume updates
+  (setq elcord--update-presence-timer nil)
+  (elcord--start-updates)
+  ;;resume timer with elapsed time
+  (setq elcord--startup-time (string-to-number (format-time-string "%s" (time-subtract nil elcord--startup-time)))))
```
Comment (chaoky) [inline elcord.el:637]: maybe extract this to a function or use let?

**Comment by enderger** [comment]
Any updates on when this will get merged?

**Comment by chaoky** [comment]
any changes needed to merge this? @Zulu-Inuoe @Mstrodl

**Comment by Lord-Valen** [comment]
Why close this?

**Comment by enderger** [comment]
It got merged into the idle branch, then into master.

**Comment by chaoky** [comment]
I didn't knew that renaming the idle branch to master would've closed this.
I have this setup on doom emacs now
```lisp
(package! elcord :recipe (:host github :repo "chaoky/elcord"))
```

**Pull request closed without merging**